tests/dialects/test_dialect.py

```diff
@@ -1097,32 +1097,51 @@ def test_order_by(self):
 
     def test_json(self):
         self.validate_all(
-            "JSON_EXTRACT(x, 'y')",
+            "JSON_EXTRACT(x, '$.y')",
             read={
-                "mysql": "JSON_EXTRACT(x, 'y')",
+                "bigquery": "JSON_EXTRACT(x, '$.y')",
+                "duckdb": "x -> 'y'",
+                "doris": "x -> '$.y'",
+                "mysql": "JSON_EXTRACT(x, '$.y')",
                 "postgres": "x->'y'",
-                "presto": "JSON_EXTRACT(x, 'y')",
-                "starrocks": "x -> 'y'",
-                "doris": "x -> 'y'",
+                "presto": "JSON_EXTRACT(x, '$.y')",
+                "sqlite": "x -> '$.y'",
+                "starrocks": "x -> '$.y'",
             },
             write={
-                "mysql": "JSON_EXTRACT(x, 'y')",
-                "oracle": "JSON_EXTRACT(x, 'y')",
-                "postgres": "x -> 'y'",
-                "presto": "JSON_EXTRACT(x, 'y')",
-                "starrocks": "x -> 'y'",
-                "doris": "x -> 'y'",
+                "bigquery": "JSON_EXTRACT(x, '$.y')",
+                "doris": "x -> '$.y'",
+                "duckdb": "x -> '$.y'",
+                "mysql": "JSON_EXTRACT(x, '$.y')",
+                "oracle": "JSON_EXTRACT(x, '$.y')",
+                "postgres": "JSON_EXTRACT_PATH(x, 'y')",
+                "presto": "JSON_EXTRACT(x, '$.y')",
+                "spark": "GET_JSON_OBJECT(x, '$.y')",
+                "sqlite": "x -> '$.y'",
+                "starrocks": "x -> '$.y'",
+                "tsql": "ISNULL(JSON_QUERY(x, '$.y'), JSON_VALUE(x, '$.y'))",
             },
         )
         self.validate_all(
-            "JSON_EXTRACT_SCALAR(x, 'y')",
+            "JSON_EXTRACT_SCALAR(x, '$.y')",
```

Comment:
can you add a test with nested paths

'$.y[0].z'

Reply:
sure

Reply:
check out https://github.com/tobymao/sqlglot/pull/2883/commits/91f073add8cf12bfb0f544dc090ecbbd83a3ded8